### PR TITLE
[MeshMovingApp] cleaning mesh-moving analysis

### PR DESCRIPTION
**Pull request by philbucher**
This cleans the `MeshMovingAnalysis` and removes the backwards-compatibility of the "old-style" project-parameters (which was anyway never really used)

@dbaumgaertner is the only user of this afaik

@dbaumgaertner I don't know, but maybe you will have to update your things after this PR.
I am doing it this way since you explicitly told me that you will act AFTER things have changed

### CODE REVIEW

**Comment by dbaumgaertner** [comment]
@philbucher I see you gentle

**Review by dbaumgaertner — COMMENTED** [review]
@philbucher I adjusted the ShapeOptApp to the new interface. @armingeiser Can you maybe double-check these changes? Thanks.

File: applications/ShapeOptimizationApplication/python_scripts/model_part_controller_factory.py [hunk 1]
```diff
@@ -41,12 +41,13 @@ def __init__(self, model_settings, model):
             "mesh_motion" : {
                 "apply_mesh_solver" : false,
                 "solver_settings" : { },
```
Comment (armingeiser) [inline applications/ShapeOptimizationApplication/python_scripts/model_part_controller_factory.py]: I think those two lines could even be removed.
```
"solver_settings" : { },
"processes" : {}
```

**Comment by armingeiser** [comment]
I am fine with this, even though i am not a big fan of updating tests of e.g. shells optimization in the ShapeOptApp in a PR that is about the MeshMovingAnalysis.

**Comment by dbaumgaertner** [comment]
True. I took the fast lane here. A separate PR would have been the better place for the tests. Sorry for the noise.

**Comment by dbaumgaertner** [comment]
Good from my side now.

**Review by armingeiser — APPROVED** [review]

**Review by dbaumgaertner — APPROVED** [review]

**Pull request merged**